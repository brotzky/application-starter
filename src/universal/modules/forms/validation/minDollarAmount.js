import numeral from 'numeral';

export default function(min) {
  return function(value) {
    return value < min
      ? {
          defaultMessage: 'cannot be less than {amount}',
          id: 'form.errors.minDollarAmount',
          values: { amount: numeral(min).format('$0,00') },
        }
      : undefined;
  };
}
