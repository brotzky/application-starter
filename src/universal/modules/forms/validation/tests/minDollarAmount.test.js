import minDollarAmount from '../minDollarAmount';

const minFiveDollars = minDollarAmount(5);

describe('[Form Validation] minDollarAmount', () => {
  describe('minDollarAmount()', () => {
    it('should return a function', () => {
      expect(minDollarAmount(5)).toMatchSnapshot();
    });

    it('should return no errors if the number passed is more than the originally specified amount', () => {
      expect(minFiveDollars(10)).toMatchSnapshot();
    });

    it('should return an error object when an amount less than specified is passed in', () => {
      expect(minFiveDollars(0)).toMatchSnapshot();
    });
  });
});
