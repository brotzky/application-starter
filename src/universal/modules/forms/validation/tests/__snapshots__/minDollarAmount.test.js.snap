// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Form Validation] minDollarAmount minDollarAmount() should return a function 1`] = `[Function]`;

exports[`[Form Validation] minDollarAmount minDollarAmount() should return an error object when an amount less than specified is passed in 1`] = `
Object {
  "defaultMessage": "cannot be less than {amount}",
  "id": "form.errors.minDollarAmount",
  "values": Object {
    "amount": "$5",
  },
}
`;

exports[`[Form Validation] minDollarAmount minDollarAmount() should return no errors if the number passed is more than the originally specified amount 1`] = `undefined`;
